### DISCUSSION TITLE
중급 프로젝트 1주차 팀회고

### DISCUSSION THREAD

**Discussion by sulyoungoh** [codeit-bootcamp-frontend/3-Weekly-Mission#337] [회고]
# 왜 회고를 해야 하나요?
- **짧은 프로젝트 기간에 팀 회고는 개인의 성장뿐 아니라, 원활한 팀 프로젝트를 위한 발판이 되어줄 거예요.**
매주 팀 단위로 회고를 하며 어떤 점이 좋았고, 어떤 점이 아쉬웠고, 아쉬웠던 부분은 어떻게 개선해볼지 함께 이야기해보면
그 다음주에는 더 나은 프로젝트를 진행할 수 있을거예요. 회고를 통해 함께 성장해보아요 🔥 

- **회고 시간만큼은 솔직한 생각, 마음을 터놓기**
솔직한 회고를 통해 다음주는 더 즐겁게 프로젝트를 진행할 수 있을거예요.

# 회고 방법
![image](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/144666341/b1f1cc83-7d24-4156-910d-3fc9b6bb7555)
회고는 **“KPT 회고”** 방식으로 진행해볼거예요. 아래 방법대로 회고해주시면 됩니다. 
- 팀원들과 모여 그 주의 팀 프로젝트 활동을 돌아보며, '좋았던 점(Keep) / 아쉬웠던 점(Problem) / 시도해볼 것(Try)'으로 항목을 나누어 회고해주세요.
    - Keep: 현재 우리팀이 잘 하고 있어서 계속 유지하면 좋을 것 같은 부분
    - Problem: 현재 우리팀이 가지고 있는 문제
    - Try: Problem에 적은 것을 해결하기 위해 우리 팀이 다음에 시도해볼만한 것 (혹은 Keep 중에서도 더 개선하거나 시도해보고 싶은 것)
- [KPT 회고 피그마 템플릿](https://www.figma.com/file/Ni3tqsOAhor97O54NZQsxc/%ED%9A%8C%EA%B3%A0-%ED%85%9C%ED%94%8C%EB%A6%BF_3%EA%B8%B0?type=whiteboard&node-id=0%3A1&t=xEi6y2qrmcRvPu67-1)을 활용해서, 팀 회고를 진행하셔도 됩니다. 단 템플릿 사용시, 팀별로 복제해서 사용해주세요. 꼭 템플릿을 사용하지 않아도 되니, 팀별로 편한 방식으로 회고를 진행해주세요. 
- **이번주 일요일까지, 팀 회고한 내용을 정리하여 댓글로 올려주세요.** 
(댓글에 예시 회고를 올려드릴테니, 참고해주세요. 피그마 템플릿으로 진행하셨다면 캡처본을 올려주셔도 돼요.)

# 참고 글
- [KPT 회고란 무엇인가?](https://brunch.co.kr/@jinha0802/35)
- [팀문화의 탄생](https://techblog.woowahan.com/2677/)

**Comment by GihoKo** [comment]
# 8팀
# **👍🏻**Keep
- 일정이 많아도 서로 협의해가면서 일정을 조율하는 모습 좋습니다
- 기초 프로젝트 경험을 기반으로 프로젝트 초반에 정해야 할 것을 수월하게 정하는 것 같아 좋습니다.
- 체계성을 가지고 준비하는 모습 좋습니다
- 다소 지난한 초반 과정이지만 다들 끝까지 집중해서 함께 하고있습니다.
- 새로운 기술을 배우고 공부해보자는 모습이 좋습니다.
- 이전 프로젝트에서 했던 방식을 자세하게 공유해주셔서 프로젝트 진행과 관련된 의사결정을 보다 쉽게 할 수 있어 도움이 되었습니다.

## **🤔**Problem
- 도전하는 부분에 있어서 조금 겁이 납니다. 열심히 물어보고 찾아보면서 학습해야 할 것 같은데 잘 적응할 수 있을지 걱정되네요.
- 새로운 기술들을 사용해야해서 러닝커브가 꽤 있습니다.
- 아직 많이 친해지지 못해서 어색한 분위기입니다.
- 의견을 잘 얘기하지 못 하는 거 같습니다.
- 팀원의 수준에 대해 잘 알지 못한 상태에서 새로운 기술들을 결정한 것 같기도 해서 걱정이 되지만, 프로젝트에 직접 적용해 보며 빠르게 성장할 것이라 생각합니다.

## **🔥**Try
- 선택과 집중이 필요한 시기인 것 같습니다. 일의 우선순위를 만들어 프로젝트를 진행함에 있어 문제가 없도록 노력해야겠습니다.
- 기술들을 공부함에 있어서 전체를 빠르게 그려보고 필요한 부분을 위주로 깊게 채워나가야 할 것 같습니다.
- 모두의 의견이 소중합니다. 어려운 점, 걱정되는 점이 있다면 지체 없이 나눠가며 진행되기를 바랍니다.
- 좀 더 의견을 자유롭게 이야기 했으면 좋겠습니다.
- 모두 동일한 기술로 프로젝트를 진행하게 될텐데 서로 유용한 내용을 자주 공유할 수 있었으면 좋겠습니다.

**Comment by nyaknya** [comment]
# 6팀
## KEEP
- 화상미팅이 아니라 대면으로 다같이 만나서 이야기해보는 시간을 가졌다.
- 이전 프로젝트 경험이 있어 시간이 오래 걸리지 않고 속전속결로 끝난것같아 좋았다.
- 쓰고 싶은 기술, 어려울 것 같은 요소를 과감하게 넣거나 빼는 의사결정 과정이 잘 됐던 것 같습니다.

## PROBLEM
- 아무래도 하루 안에 다 준비해야한다는 압박이 있다 보니 놓친 요소가 몇 있는 것 같습니다.
- 새로운 기술을 쓰기로 한 것이 있어서, 해당 부분 러닝커브가 걱정됩니다.
- 일정에 대한 부분이 조금 걱정됩니다.

## TRY
- 매일 데일리 스크럼을 진행하니, 그때 놓친 것이나 체크해야될게 있다면 주제를 꺼내 잘 조율해보면 될 것 같습니다. 
- 지금도 서로 머뭇거리는 분위기는 아니라고 생각하지만, 좀 더 편하게 이야기해봅시다!

**Comment by soyxxn** [comment]
🤓 3팀
=============
중급 프로젝트 1주차 만족도 ⭐2.6 / 5
-------------

### KEEP
-  다들 적극적으로 다양하게 의견을 제시하는 점이 좋았습니다.
- 딱딱한 회의 분위기가 아닌 자유롭게 조율해가는 분위기가 좋았고, 빠르게 진행되는데도 순조로운 방향대로 흘러가는 점이 좋았습니다.
- 적극적으로 참여하는 모습에 동기부여가 되었습니다.
- 정해진대로 고정시킨다기보다 시도해보고 유동적으로 움직이자는 분위기가 효율적으로 느껴졌습니다.
- 팀미팅 마무리에 다음날 미팅 때 나누어 볼 이야기까지 미리 정해두어서 그것을 이정표로 삼을 수 있어 좋았습니다.
- 회의 내용이 잘 정리된 템플릿을 만들어주셔서 두고두고 참고할 수 있는 점이 좋았습니다.

### PROBLEM
- 프로젝트를 위해 팀미팅 시간을 9시로 변경했는데 불규칙한 수면시간을 잘 조절해야 할 것 같습니다.
- 팀 컨벤션을 정한지 얼마 되지 않았기 때문에 익숙해지는 시간이 필요할 것 같습니다.
- 지금까지 과제를 하며 정해진 나름의 스타일과는 거의 반대로 이야기가 흘러가서 부정적인 사고의 연쇄로 이어진 것 같습니다.
- 코어타임을 디스코드 들어와있기 까지만 정해둔 게 두루뭉술한 것 같아 좀 더 명확한 방침을 두고 잘 지켜나가면 좋을 것 같습니다.
- 다들 너무 착하셔서 돌려말하는 경향으로 회의시간이 길어지는 것 같습니다. 회의시간이 너무 길어지면 공부할 시간이 줄어들 수 있기 때문에 줄일 수 있는 시간은 최대한 줄여나가고, 길어질 경우에는 중간중간 리프레시 하는 시간이 필요할 것 같습니다.
- 부족한 게 많아 팀미팅 때 적극적으로 참여하지 못한 것 같아 아쉽습니다.
- 다른 분들 말을 자르는 것 같아 조금 더 신경써야 할 것 같습니다.

### TRY
- 가능하면 낮 시간을 활용하여 모든 것을 집중해서 처리하고 수면시간이 되면 무조건 눕기
- 일찍 자고 일찍 일어나는 습관 들이기
- 팀미팅이 끝나면 바로 작업하고, 코어 타임을 꼭 지키기
- 커밋 전, PR 올리기 전에 팀 컨벤션을 준수하여 리팩토링 후 올리기
- 일정을 잘 지키고 구체적인 작업 계획 세우기
- 다른 사람 의견을 끝까지 듣고 고려하기
- 모르는 점은 찾아볼 때도, 질문할 때도 용이하게 정리를 잘 해두기

**Comment by yulmai999** [comment]
# 5팀

# Keep

- 피그마로 공통 컴포넌트를 잘 나누었다.
- 회의가 바로바로 잘 되고 의견공유가 활발하여 좋았다.
- 프로젝트에 대해 체계적으로 접근하는 방식이 좋았다.
- 프로젝트 기초를 다지는데있어서 의견교환이 잘되었고 체계적이라좋았다.

## Problem

- 리액트 쿼리, 커스텀 훅, 테윌윈드에 대한 공부가 부족하다.
- 프로젝트에 들어가지 않아 어떻게 하면 좋을지 확신이 안든다.
- 앞으로 어떤 변수들이 생길지 아직 모르겠다.
- 새로운 기술스택을 많이사용해서 공부하고 적용하는데 시간이 많이걸릴거같다

## Try

- 공부를 계속하고, 익숙하지 않아도 계속 사용해봐야 될 것 같다.
- 개발을 해 나가면서 애매모호한 부분을 해결해 나가자.
- 문제가 생기더라도 팀원들과 소통하며 잘 풀어나가보자.
- 새로운 기술을 적용할때 서로 정보공유와 의사소통을 적극적으로 하면좋을거같다

---

**Comment by mingzzi96** [comment]
# 7팀

### KEEP
- 초기 셋팅 무사히 마무리하여 공유하였다.
- 프로젝트를 통해 머리로만 알고 있던것을 실제로 써보니 많은 도움이 되었다.
- 코어타임을 정해서 집중하는 시간을 가진게 좋았다.

### PROBLEM
- 셋팅 관련 주석 설명이 부족했던 것 같다.
- 실제로 만들다보니 R&R분배에서 모호하게 넘어간 부분이 있다고 생각이 들었다.
- 초반 세팅할때 관련지식이 부족하여 알아듣느데 시간이 필요했던것 같다.

### TRY
- 주석 달기를 생활화하고 자세히 써야겠다.
- 팀채널 활용해서 적극적인 의견 공유 및 피드백
- 초반 세팅 주석을 활용해보고 팀과의 의사소통으로 이해해보자

**Comment by DanteSnow** [comment]
## 🔥2팀🔥

## Keep

- 팀원들 모두가 다른 팀원을 먼저 배려해주려는 분위기가 정말 좋았습니다.
- 의견 대립이 생겼을 때에도 하나하나 귀 기울여주고, 반영하려고 노력하는 분위기가 정말 좋았습니다.
- 도전적이고 열정적이면서도 부드러운 팀을 만난 것 같아서 어떤 일이든 함께 해낼 수 있다는 확신이 들었습니다.
- 다들 정말 중급 프로젝트 시작 전부터 의지가 눈에 보일 정도로 열정적이여서 저도 같이 힘이 났던 것 같습니다. 이번 프로젝트에 있어 기본적인 규칙과 컨벤션 룰을 정하는데 수월하게 진행되었던 점과 의견 차이가 날 수 있었던 역할 분담에 있어서도 팀원들을 먼저 배려하는 모습들이 좋은 팀을 만났다고 느낄 수 있었습니다. 또한 모든 팀원들이 새로운 것에 도전하는데에 있어 정말 열정적인 것도 좋았던 것 같습니다.
- 각자의 역할을 나누고 일을 진행함에 있어서 효율적이고 체계적인 것 같아서 좋습니다.
- 컨벤션과 규칙을 정하는데에 있어 막힘없이 정해져 수월하게 진행되는 느낌이 좋습니다.
- 새로운 라이브러리를 사용해서 경험할 수 있어서 좋습니다.

## Problem

- 익숙한 라이브러리가 아닌 새로운 라이브러리를 학습하면서 진행해야해서 시간이 조금 오래 걸리지만 아직까지는 괜찮음.
- 새로운 방식들을 학습하면서 진행하다보니 다소 진행하는데에 있어 속도면에 있어 느렸던 것 같습니다. 그 외에는 아직 초반이라 별 문제가 없는 것 같습니다.
- 제가 이사를 하게 되어서 팀 프로젝트에 온전히 에너지를 쏟지 못한다는 것이 유일한 문제인 것 같습니다.

## Try

- 얼른 기본 요구사항을 끝내고 새로운 기능을 넣자!
- 가장 중요한 것은 부딪히면서 배우는 것이라고 생각합니다. 초반에 정했던 우선순위를 기준으로 빠르게 진행하고 부가적인 디자인과 기능들은 점차 완성하는 방향으로 진행하면 될 것 같습니다.
- 팀 프로젝트를 위해서 이사를 빠르게 잘 진행해보겠습니다!

**Comment by 2zzzyoung** [comment]
# 4팀🐻‍❄️
## Keep
- PR 승인이 빠르다.
- 알아서 기능 개발을 잘 하고 있는 것 같다.
- 라이브러리 등을 사용하는데 거리낌이 없다.
- 저번 프로젝트에서 경험해보지 못한 방식을 경험해볼 수 있어서 좋다.
- 소통이 잘 되고 질문이나 답변을 많이 한다.
- 속전속결로 분배하여 빠르게 시작한 점이 좋았다.
- 서로 모르는 부분, 겹치는 부분(공통컴포넌트), 폴더구조 맞추기 등 상호 이야기가 필요한 문제를
  실시간으로 대화하며 바로 바로 확인, 수정하는 것이 잘 이루어지고 있고 더 활발하게 유지되었으면 좋겠다.

## Problem
- PR 승인은 빠르지만 내용 확인을 잘 안 하는 것 같다. 그체적 리뷰가 있었으면 좋겠다.
- 실시간으로 채팅방에서 현재 개발 상황 이야기를 했으면 좋겠다. (다같이)
- 맡은 기능을 다른 팀원과 연결해야 하는 부분이 있을 때 내가 빠르게 만들어내지 못해서 걱정이 된다.
- 원래 예정된 프로젝트 기간 전부터 시작해서 그 주 강의 듣는게 좀 밀렸지만 아직까진 괜찮다.
- 힘든게 있으면 서로 더 적극적으로 어필하는게 좋을거 같다.

## Try
- 기능 개발이나 요구사항 외에 시도해볼만한 것들을 해봤으면 좋겠다. (개인적으로 적용하고 팀에 공유하는 방식으로. Ex) git actions, test, 기능 구현과 리팩토링 후 코드리뷰 등… )
- 팀원 분들께 많이 물어보며 배워야겠다.
- 폴더나 파일을 만들 때 규칙을 더 개선하면 좋을 것 같다.
- 하다보면 해결할 수 있을 것 같다.

**Comment by jihee1103** [comment]
### 1팀 
## Keep

- 적극적으로 질문을 하고 대답을 다들 잘 해준다.
- 화목한 분위기
- 회의를 열심히 참여한다. 
- 의사표현이 자유롭고 소통이 잘된다.
- 코어타임이 정해져있어서 늘어지지 않는 점이 좋다.
- 역할 나누기
- 코드 리뷰
- 깃허브 이슈랑  pr 활용,  리뷰 적극적으로 너무 잘한다.
- 새벽까지 열심히 하는 모습
- 깃허브 이슈를 잘 활용하고있다.
- 의견공유가 잘되고, 문제를 다같이 해결하려고한다.
- 주석 완전 열심히 달아주시는 모습 왕칭찬드립니다.
- 팀장님의 솔선수범 아주 칭찬드립니다. 피그잼 처음 알았는데 너무 좋고요 피그마 복사해온 거 아주아주 편하고 짱!

## Problem

- 프로젝트 세팅이 중간중간 바뀌었다.
- 내가 너무 적은 파트를 맡은 거 같아서 마음이 불편하다.
- 세부적인 컨벤션 조정이 안되어 이후에 추가적인 수정 사항이 발생
- 컨벤션이나 초기세팅에서 디테일한 부분을 신경쓰지 못했다.
- 초기 세팅에서 부족한 부분이 있었다.

## Try

- 시간 되면 상태관리 사용해보고싶다!
- 서로서로 열심히 도와서 만들기
- 코드리뷰를 어떻게 해야할지 몰라서 잘 못했는데 사소한거라도  리뷰해보기!
- 어렵거나 막히는 부분 적극적으로 공유하기
- 다른 사람이 읽기 어려울 것 같은 코드는 주석으로 설명 열심히 해주기
- 각자가 해야할 부분을 명확하게 나누고 좀 더 강제적인 폴더명 규칙 정하기

**Comment by 2zzzyoung** [comment]
## 4팀🐵
### Keep
- 다들 의견 수용을 잘 해주신다.
- 다같이 디스코드 음성채팅방에 모여서 작업해 소통이 잘 되었던 것 같다.
### Problem
- 기본 동작 버그 부분을 확인하지 않은 점
- 발표 준비를 다같이 하지 않은 것
- 에러가 많아서 없애기 힘들었다
### Try
- 그럼에도 불구하고 잘 끝냈다
- 에러 방지를 위해 타입 지정을 그때그때 바로 하기
